packages/zoe/src/contractFacet/offerSafety.js

```diff
@@ -2,25 +2,41 @@
 
 import { AmountMath } from '@agoric/ertp';
 
+const { entries } = Object;
+
 /**
- * Helper to perform satisfiesWant and satisfiesGive. Is
- * allocationAmount greater than or equal to requiredAmount for every
- * keyword of giveOrWant?
+ * Helper to perform satisfiesWant and satisfiesGive. How many times
+ * does the `allocation` satisfy the `giveOrWant`?
  *
  * @param {AmountKeywordRecord} giveOrWant
  * @param {AmountKeywordRecord} allocation
+ * @returns {bigint}
  */
 const satisfiesInternal = (giveOrWant = {}, allocation) => {
-  const isGTEByKeyword = ([keyword, requiredAmount]) => {
-    // If there is no allocation for a keyword, we know the giveOrWant
-    // is not satisfied without checking further.
+  /** @type {bigint | undefined} */
+  let multiples; // undefined represents infinity
+  for (const [keyword, requiredAmount] of entries(giveOrWant)) {
     if (allocation[keyword] === undefined) {
-      return false;
+      return 0n;
     }
     const allocationAmount = allocation[keyword];
-    return AmountMath.isGTE(allocationAmount, requiredAmount);
-  };
-  return Object.entries(giveOrWant).every(isGTEByKeyword);
+    if (!AmountMath.isGTE(allocationAmount, requiredAmount)) {
+      return 0n;
+    }
+    if (typeof requiredAmount.value !== 'bigint') {
+      multiples = 1n;
```

Comment:
too much inconsistency between immediate returns and setting `multiples`.  Here I'd `return 1n`.

Reply:
The `multiples` here are being set within the loop, whereas the `multiples` variable is being set outside the loop. If on one iteration this condition applies, it'll set `multiples` to `1n`. If on a later iteration the condition above it applies, then the function as a whole will return `0n`. If we returned here, the function as a whole would return `1n` instead, which would be wrong.